pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -254,3 +263,12 @@ func (t *TopologyCache) getAllocations(numEndpoints int) map[string]Allocation {
 
 	return allocations
 }
+
+func hasExcludedLabels(labels map[string]string) bool {
```

Comment:
just curious, is there  informer lister/indexer trickery to do this?

Reply:
The problem is that I don't think we can exclude these for the entire EndpointSlice controller, we just want to exclude them from this specific topology calculation.

Reply:
I mean here , "list all nodes but this ones"
https://github.com/kubernetes/kubernetes/blob/0ec2aaee473ef6256361bfa1aa044909dd0132fe/pkg/controller/endpointslice/endpointslice_controller.go#L539-L542

it seems there is no negative matching for label selector, right?

Reply:
Good point, yeah, I don't think it's possible to do a negative match with a selector like that.